### PR TITLE
Some fixes and features

### PR DESCRIPTION
**Pull request by Iksxsuzil**
1. Cover size option (default to large)
1. Fetch allmusic covers
1. Fetch deezer covers
1. Fix junodownload cover fetching
1. Auto select allmusic and deezer

### CODE REVIEW

File: pth_yadg.user.js [hunk 1]
```diff
@@ -79,9 +79,15 @@ function fetchImage(target, callback) {
 				method: 'GET',
 				url: 'https://itunes.apple.com/lookup?id=' + id + '&country=' + country,
 				onload(response) {
-					if (response.status === 200) {
+					if (response.status === 200) {                  
 						const data = JSON.parse(response.responseText);
-						const hires = data.results[0].artworkUrl100.replace('100x100bb', '100000x100000-999');
+						var hires;
```
Comment (SavageCore) [inline pth_yadg.user.js]: Use `let` or `const`

File: pth_yadg.user.js [hunk 2]
```diff
@@ -79,9 +79,15 @@ function fetchImage(target, callback) {
 				method: 'GET',
 				url: 'https://itunes.apple.com/lookup?id=' + id + '&country=' + country,
 				onload(response) {
-					if (response.status === 200) {
+					if (response.status === 200) {
```
Comment (SavageCore) [inline pth_yadg.user.js]: Trim trailing spaces

**Comment by Iksxsuzil** [comment]
Thank you for help with XO. I checked project before commit. I left choice for users, the default is maximum size. 
chrome_2018-11-29_01-06-26

Example large cover:
- 8 MB https://f4.bcbits.com/img/a0523520394_0.jpg
- 9 MB https://f4.bcbits.com/img/a0245417581_0.jpg 
- 8.5 MB https://f4.bcbits.com/img/a3559459035_0 
- 5.9 MB https://is3-ssl.mzstatic.com/image/thumb/Music128/v4/52/76/d9/5276d9af-383f-ed42-d708-38129ef585a7/source/100000x100000-999.jpg 
- 5 MB https://is3-ssl.mzstatic.com/image/thumb/Music118/v4/5a/c5/3f/5ac53f2f-02b0-4729-e9ce-83994da4ec5b/source/100000x100000-999.jpg 
- 11 MB o.O https://is1-ssl.mzstatic.com/image/thumb/Music128/v4/4b/ff/25/4bff25a1-127a-0784-5b29-bdfd95dfc7b2/source/100000x100000-999.jpg 
- 5.9 MB https://is3-ssl.mzstatic.com/image/thumb/Music128/v4/52/76/d9/5276d9af-383f-ed42-d708-38129ef585a7/source/100000x100000-999.jpg 
- 5 MB https://is3-ssl.mzstatic.com/image/thumb/Music118/v4/5a/c5/3f/5ac53f2f-02b0-4729-e9ce-83994da4ec5b/source/100000x100000-999.jpg

**Comment by SavageCore** [comment]
Looks good. Thanks!

**Pull request merged**